File: nodejs/http-method-identifier/handler.js
export const handler = async (req, res) => {
  try {
    // Determine the HTTP method
    const method = req.method;

    let message;
    if (method === "GET") {
      message = "GET method called.";
    } else if (method === "POST") {
      message = "POST method called.";
    } else if (method === "PUT") {
      message = "PUT method called.";
    } else if (method === "DELETE") {
      message = "DELETE method called.";
    } else if (method === "PATCH") {
      message = "PATCH method called.";
    } else {
      message = "Unsupported method.";
    }

    // Set the response JSON
    const responseJson = { message };

    // Set the response headers
    res.setHeader("Content-Type", "application/json");

    // Send the response JSON
    res.end(JSON.stringify(responseJson));
  } catch (error) {
    // Handle errors
    console.error(error);
    // Send an error response if needed
    res.statusCode = 500;
    res.setHeader("Content-Type", "text/plain");
    res.end("Internal Server Error");
  }
};

File: python/http-method-identifier/handler.py
from flask import jsonify
import logging

def handler(request):
    try:
        # Determine the HTTP method
        method = request.method

        if method == 'GET':
            message = 'GET method called.'
        elif method == 'POST':
            message = 'POST method called.'
        elif method == 'PUT':
            message = 'PUT method called.'
        elif method == 'DELETE':
            message = 'DELETE method called.'
        elif method == 'PATCH':
            message = 'PATCH method called.'
        else:
            message = 'Unsupported method.'

        # Set the response JSON
        response_json = {'message': message}

        # Return the response JSON
        return jsonify(response_json)
    except Exception as error:
        logging.error(error)
        # Send an error response if needed
        return jsonify({
            "statusCode": 500,
            "headers": {"Content-Type": "text/plain"},
            "message": "Internal Server Error"
        })
